plutus-tx/src/Language/PlutusTx/Prelude.hs

```diff
@@ -58,37 +68,86 @@ traceH :: Q (TExp (String -> a -> a))
 traceH = [|| \str a -> $$(trace) ($$(toPlutusString) str) a||]
 
 -- | Logical AND
+--
+--   >>> $$([|| $$(and) True False ||])
+--   False
+--
 and :: Q (TExp (Bool -> Bool -> Bool))
 and = [|| \(l :: Bool) (r :: Bool) -> if l then r else False ||]
 
 -- | Logical OR
+--
+--   >>> $$([|| $$(or) True False ||])
+--   True
+--
 or :: Q (TExp (Bool -> Bool -> Bool))
 or = [|| \(l :: Bool) (r :: Bool) -> if l then True else r ||]
 
 -- | Logical negation
+--
+--   >>> $$([|| $$(not) True ||])
+--   False
+--
 not :: Q (TExp (Bool -> Bool))
 not = [|| \(a :: Bool) -> if a then False else True  ||]
 
 -- | The smaller of two `Int`s
+--
+--   >>> $$([|| $$(min) 10 5 ||])
+--   5
+--
 min :: Q (TExp (Int -> Int -> Int))
 min = [|| \(a :: Int) (b :: Int) -> if a < b then a else b ||]
 
 -- | The larger of two `Int`s
+--
+--   >>> $$([|| $$(max) 10 5 ||])
+--   10
+--
 max :: Q (TExp (Int -> Int -> Int))
 max = [|| \(a :: Int) (b :: Int) -> if a > b then a else b ||]
 
+-- | Check if a `Maybe` @a@ is @Just a@
+--
+--   >>> $$([|| $$(isJust) Nothing ||])
+--   False
+--   >>> $$([|| $$(isJust) (Just "plutus") ||])
+--   True
+--
 isJust :: Q (TExp (Maybe a -> Bool))
-isJust = [|| \(m :: Maybe a) -> case m of { Just _ -> True; _ -> False; } ||]
-
+isJust = [|| \m -> case m of { Just _ -> True; _ -> False; } ||]
+
+-- | Check if a `Maybe` @a@ is @Nothing@
+--
+--   >>> $$([|| $$(isNothing) Nothing ||])
+--   True
+--   >>> $$([|| $$(isNothing) (Just "plutus") ||])
+--   False
+--
 isNothing :: Q (TExp (Maybe a -> Bool))
-isNothing = [|| \(m :: Maybe a) -> case m of { Just _ -> False; } ||]
-
+isNothing = [|| \m -> case m of { Just _ -> False; _ -> True; } ||]
+
+-- | The maybe function takes a default value, a function, and a `Maybe` value. 
```

Comment:
I wonder whether it's easier to just document these by referring to the normal functions which they're ports of?

Reply:
yeah that's probably better than copying the text from the originals

Reply:
I added links whereever the explanation was longer than one sentence.